components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -304,15 +312,21 @@ public class LegacyJobOrchestrator : IJobOrchestrator, IPublisherConfigServices,
                                         }
                                     }
                                 }
-                                _availableJobs = availableJobs;
-                                _assignedJobs = assignedJobs;
+                            }
+
+                            _publishedNodesEntrys.Clear();
+                            if (entries != null) {
 
-                                _publishedNodesEntrys.Clear();
                                 _publishedNodesEntrys.AddRange(entries);
                             }
-                            else {
-                                _availableJobs.Clear();
-                                _assignedJobs.Clear();
+
+                            _lockJobs.Wait();
```

Comment:
Reading values of `_assignedJobs` should also be under `_lockJobs`, such as on line `303` above:

```
foreach (var assignedJob in _assignedJobs) {
```

Reply:
Nice catch. I tried to simplify the number of locks, but obviously missed one usecase

Reply:
@cristipogacean, access to `_availableJobs` and `_assignedJobs` seem to still not be under `_lockJobs` lock in `TriggerAgentConfigUpdate()` method.

Reply:
@karok2m , agreed and fixed.